contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,81 @@
+# Priority in ResourceQuota
+
+@resouer
+
+Aug 2017
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota apply to pod](#resourcequota-apply-to-pod)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, three sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies a priority class name.
+* Incorporate priority in quota checks.
+* ResourceQuota admission controller to reject ResourceQuota based on its priority class.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this is implemented in: #45610)
+* Incorporate priority in pod scheduling logic.
+
+## Background 
+
+Since we already have [priority filed in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. So it is nature to make ResourceQuota apply to Pods based on its priority class. In order to implement this, we need to add a priority filed to ResourceQuota definition.
+
+## Overview 
+
+This design doc introduces the motivation of adding priority filed in ResourceQuota and
+how this priority impacts ResourceQuta applying to Pod. 
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota apply to pod 
+
+The design will be like:
+
+```
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // If specified, indicates this ResourceQuota applies to that priority class
+  // If not specified, this ResourceQuota applies to all priority classes.
+  // +optional
+  PriorityClassName string
+  // The priority value. Various system components use this field to find the
+  // priority of the ResourceQuota. When Priority Admission Controller is enabled, it
+  // prevents users from setting this field. The admission controller populates
+  // this field from PriorityClassName.
+  // +optional
+  Priority *int32
+}
+```
+
+* One could expect when `PriorityClassName` field is set, it indicates that this quota applies to Pod with same priority class. 
```

Comment:
Yes. PriorityClassName is used in our UI as a string and easy to remember name. Integer priority is used by our internal components. So, we should let users set PriorityClassName.